src/sentry/api/endpoints/group_hashes.py

```diff
@@ -32,15 +41,30 @@ def get(self, request, group):
         :auth: required
         """
 
-        queryset = GroupHash.objects.filter(
-            group=group.id,
+        aggregations = [
+            ('argMax(event_id, timestamp)', None, 'event_id')
+        ]
+
+        filter_keys = {
+            'project_id': [group.project_id],
+            'group_id': [group.id]
+        }
+
+        data_fn = partial(
+            lambda *args, **kwargs: raw_query(*args, **kwargs)['data'],
+            start=datetime.utcfromtimestamp(0),  # will be clamped to project retention
+            end=datetime.utcnow(),  # will be clamped to project retention
+            aggregations=aggregations,
+            filter_keys=filter_keys,
+            selected_columns=['primary_hash'],
+            groupby=['primary_hash'],
+            referrer='api.group-hashes',
         )
 
         return self.paginate(
             request=request,
-            queryset=queryset,
-            order_by='id',
-            on_results=lambda x: serialize(x, request.user),
+            on_results=lambda results: self.handle_results(results, group.project_id),
```

Comment:
nit: It'd be slightly cleaner (to me, at least) to reverse the argument order of `handle_results` and just partial `group.project_id` onto `self.handle_results`.